TESTS/mbed_hal/lp_us_tickers/main.cpp

```diff
@@ -302,9 +307,23 @@ void ticker_increment_test(void)
         uint32_t next_tick_count = base_tick_count;
         uint32_t inc_val = 0;
 
-        while (next_tick_count == base_tick_count) {
+        while (inc_val < 100) {
             next_tick_count = count_ticks(NUM_OF_CYCLES + inc_val, 1);
-            inc_val++;
+
+            if (next_tick_count == base_tick_count) {
+                /* Same tick count, so increase num of cycles. */
+                inc_val++;
+            } else {
+                /* It is possible that the difference between base and next
+                 * tick count on some platforms is greater that 1, in this case we need
+                 * to repeat counting with the same number of cycles.
+                 * In cases if difference is exactly 1 we can exit the loop.
+                 */
+                if (next_tick_count - base_tick_count == 1 ||
```

Comment:
Isn't that what this test is validating in the first place? If the core is too slow, you could reduce NUM_OF_CYCLES instead.

Reply:
Fixed.